src/mgmt/srv_drpc.c

```diff
@@ -1147,8 +1146,23 @@ add_props_to_resp(daos_prop_t *prop, Mgmt__PoolGetPropResp *resp)
 			if (resp_props[j]->strval == NULL)
 				D_GOTO(out, rc = -DER_NOMEM);
 		} else if (daos_prop_has_ptr(entry)) {
-			D_ERROR("pointer-value props not supported\n");
-			D_GOTO(out, rc = -DER_INVAL);
+			switch (entry->dpe_type) {
+			case DAOS_PROP_PO_SVC_LIST:
+				if (entry->dpe_val_ptr == NULL) {
+					D_ERROR("svc rank list unset\n");
+					D_GOTO(out, rc = -DER_INVAL);
```

Comment:
nit: I guess I hear conflicting advice about the use of D_GOTO, I thought the current guidance was to avoid using it.

Reply:
I was never totally clear on that myself. I've been continuing to use it where it's convenient.